src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -268,10 +270,11 @@ public void execute(BuildResult buildResult) {
             }
 
             if (isGitHubActions(providers)) {
+                String gitHubUrl = envVariable("GITHUB_SERVER_URL", providers).orElse("https://github.com");
```

Comment:
In which scenarios is the GITHUB_SERVER_URL not present when invoked from a GH action (and the need for a fallback)?

Reply:
I could not find any specific scenarios where we would expect them to not be present when invoked from a GitHub Action, but as @jprinet mentioned it *may* be possible given some older runners or on-prem runners. But maybe a fallback isn't the correct way to handle those environment variables missing? Not generating the link may be more correct.

Reply:
I would not add the fallback since you cannot be sure the build runs on github.com, so worst case, the generated links are wrong. IMO, no link is better than a wrong link.

Reply:
Done.